release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,238 @@
+module openconfig-hashing {
+    yang-version "1";
+
+    // namespace
+    namespace "http://openconfig.net/yang/hashing";
+
+    prefix "oc-hashing";
+
+    // import some basic types
+    import openconfig-extensions { prefix oc-ext; }
+
+    // meta
+    organization "OpenConfig working group";
+
+    contact
+      "OpenConfig working group
+      netopenconfig@googlegroups.com";
+
+    description
+      "Model for managing hashing policies that would be referenced by the
+      interfaces model.";
+
+
+    oc-ext:openconfig-version "0.1.0";
+
+    revision "2023-08-08" {
+        description "Initial hashing model.";
+        reference "0.1.0";
+    }
+
+    // OpenConfig specific extensions for module metadata.
+    oc-ext:regexp-posix;
+    oc-ext:catalog-organization "openconfig";
+    oc-ext:origin "openconfig";
+
+    typedef hash-field-type-ipv4 {
+        type enumeration {
+            enum NONE {
+                description "Do not use any IPv4 fields in the
+                    calculation of the hash.";
+            }
+            enum SRC_ADDRESS {
+                description "Use only the IPv4 source address in the
+                    calculation of the hash.";
+            }
+            enum DST_ADDRESS {
+                description "Use only the IPv4 destination address in the
+                    calculation of the hash.";
+            }
+            enum BOTH {
+                description "Use both the IPv4 source and
+                    destination addresses in the calculation of the
+                    hash.";
+            }
+            enum TARGET_DEFINED {
```

Comment:
How does one test this configuration parameter?

Reply:
A test would require target specific knowledge to create a test for this particular enum.  Use of this enum is expected when the target contains some default or hardcoded selection of fields which are not described by the model.  

For example, an asic may select only a portion of the bits in the src and dest ip address fields and port number fields.  Which bits are used may not be exposed and may not be configurable by the user, by the NOS or by the ASIC SDK.  Yet they are still "target defined" and don't match the other enums presented here.  

If a target defined mode isn't available on an implementation, then the implementation should return an UNIMPLEMENTED (12) error.

Reply:
We can update the description for this enum in #959

Reply:
Continue the discussion here as desired: https://github.com/openconfig/public/pull/959/files#r1337998777